tests/SelfTest/IntrospectiveTests/TestCaseInfoHasher.tests.cpp

```diff
@@ -0,0 +1,41 @@
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/catch_test_case_info.hpp>
+#include <catch2/internal/catch_test_case_info_hasher.hpp>
+
+static constexpr Catch::SourceLineInfo dummySourceLineInfo = CATCH_INTERNAL_LINEINFO;
+
+TEST_CASE( "TestCaseInfoHasher produces equal hashes." ) {
+    SECTION( "class names and names and tags are equal." ) {
+        Catch::TestCaseInfo testCase1("", {"name", "[.magic-tag1]"}, dummySourceLineInfo);
+        Catch::TestCaseInfo testCase2("", {"name", "[.magic-tag1]"}, dummySourceLineInfo);
+
+        Catch::TestCaseInfoHasher h;
+        CHECK(h(testCase1) == h(testCase2));
+    }
+}
+
+TEST_CASE( "TestCaseInfoHasher produces different hashes." ) {
+    SECTION( "class names are equal, names are equal but tags are different." ) {
+        Catch::TestCaseInfo testCase1("", {"name", "[.magic-tag1]"}, dummySourceLineInfo);
+        Catch::TestCaseInfo testCase2("", {"name", "[.magic-tag2]"}, dummySourceLineInfo);
+
+        Catch::TestCaseInfoHasher h;
+        CHECK(h(testCase1) != h(testCase2));
+    }
+
+    SECTION( "class names are equal, tags are equal but names are different" ) {
+        Catch::TestCaseInfo testCase1("", {"name1", "[.magic-tag]"}, dummySourceLineInfo);
+        Catch::TestCaseInfo testCase2("", {"name2", "[.magic-tag]"}, dummySourceLineInfo);
+
+        Catch::TestCaseInfoHasher h;
+        CHECK(h(testCase1) != h(testCase2));
+    }
+
+    SECTION( "names are equal, tags are equal but class names are different" ) {
+        Catch::TestCaseInfo testCase1("class1", {"name", "[.magic-tag]"}, dummySourceLineInfo);
+        Catch::TestCaseInfo testCase2("class2", {"name", "[.magic-tag]"}, dummySourceLineInfo);
+
+        Catch::TestCaseInfoHasher h;
+        CHECK(h(testCase1) != h(testCase2));
+    }
+}
```

Comment:
This looks pretty reasonable.

You should also add a test that the hasher can be seeded (and that the same `TestCaseInfo` produces different results for differently seeded hashers).

I would also nitpick different section names, but those are just nitpicks and these are okay.

Reply:
When i understand you right it shall be possible to seed or not seed the TestCaseInfoHasher, thats what the latest commit contains.

The specification of FNV-1a https://datatracker.ietf.org/doc/html/draft-eastlake-fnv-17.html states that "almost any offset_basis will serve so long as it is non-zero". Currently one can pass a seed of zero to TestCaseInfoHashers constructor which we could prevent with for example a fallback but i don't think that's really an issue.

Reply:
Right, I wasn't clear enough originally.

The original code used a known good seed for the hash, but
then mixed in seed from Catch2 to randomize the result. Let's call these
the hash-seed (the `hash_t hash = 14695981039346656037u` in the original
code), and Catch2-seed (what Catch2 passes into the hasher from user config).

The reason it is done like this is that the test case hashing is used
to implement `--order rand` in a stable way. The test cases are hashed,
the hashes are sorted, and the resulting order is taken as the order
of test cases to be run.

So what should be done is that the algorithm keeps the known good hash-seed
(ideally the same one), and mixes in the Catch2-seed at the end, the way
the original implementation used to do. This includes the shortened
multiplication, as it improves the scattering of similar test cases.

Reply:
Thank you for the detailed explanation, i read it yesterday and was about to commit almost the same as you did today but hesitated and felt i have to think a bit about the constructor-with-no-parameter case that you removed which is fine. :-)
The latest commit adapts the unit tests and the return value type.